pkg/skaffold/build/gcb/jib.go

```diff
@@ -21,26 +21,35 @@ import (
 
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/jib"
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/latest"
+	"github.com/pkg/errors"
 	cloudbuild "google.golang.org/api/cloudbuild/v1"
 )
 
-func (b *Builder) jibMavenBuildSpec(artifact *latest.JibMavenArtifact, tag string) cloudbuild.Build {
-	return cloudbuild.Build{
-		Steps: []*cloudbuild.BuildStep{{
-			Name:       b.MavenImage,
-			Entrypoint: "sh",
-			Args:       fixHome("mvn", jib.GenerateMavenArgs("build", tag, artifact, b.skipTests, b.insecureRegistries)),
-		}},
+func (b *Builder) jibBuildSpec(artifact *latest.Artifact, tag string) (cloudbuild.Build, error) {
+	t, err := jib.DeterminePluginType(artifact.Workspace, artifact.JibArtifact)
+	if err != nil {
+		return cloudbuild.Build{}, err
```

Comment:
Q: why an empty `cloudbuild.Build{}` instead of nil?

Reply:
It's a value and not a reference, so can't return nil.